packages/apex-lsp-testbed/src/test-utils/traceDataUtils.ts

```diff
@@ -0,0 +1,180 @@
+/*
+ * Copyright (c) 2025, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the
+ * repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+/**
+ * Detects if a URI is a Windows file URI.
+ * @param uri - The URI to check
+ * @returns True if it's a Windows file URI
+ */
+const isWindowsFileUri = (uri: string): boolean =>
+  uri.match(/^file:\/\/[A-Za-z]:/) !== null;
+
+/**
+ * Extracts the file path from a file URI, handling both Windows and Unix formats.
+ * @param uri - The file URI
+ * @returns The file path
+ */
+const extractFilePathFromUri = (uri: string): string => {
+  if (isWindowsFileUri(uri)) {
+    // Windows: file://D:\path\to\file -> D:\path\to\file
+    return uri.replace(/^file:\/\//, '');
+  } else {
+    // Unix: file:///path/to/file -> /path/to/file
+    return uri.replace(/^file:\/\/\//, '/');
+  }
+};
+
+/**
+ * Normalizes URIs in the trace data to use relative paths instead of absolute paths.
+ * This makes the test data portable across different environments and contributors.
+ * @param data - The trace data to normalize
+ * @returns Normalized trace data
+ */
+export const normalizeTraceData = (
+  data: Record<string, any>,
+): Record<string, any> => {
+  const normalizedData = JSON.parse(JSON.stringify(data));
```

Comment:
how exactly does stringifying and then parsing that string normalize the data versus just leaving data as-is?